cmd/auth/login.go

```diff
@@ -3,26 +3,279 @@
 func authLoginRunE(cmd *cobra.Command, args []string) error {
-	// l := logger.Get()
-	// l.Debug().Msgf("Auth Login Subcommand Called.")
+	ctx := context.Background()
+
+	// Get current profile name for messaging
+	profileName, err := profiles.GetOptionValue(options.RootActiveProfileOption)
+	if err != nil {
+		return fmt.Errorf("failed to get active profile: %w", err)
+	}
+
+	// Check if any authentication method flag was explicitly provided
+	deviceCodeFlag := cmd.Flag(options.AuthMethodDeviceCodeOption.Flag.Name)
+	clientCredentialsFlag := cmd.Flag(options.AuthMethodClientCredentialsOption.Flag.Name)
+	authCodeFlag := cmd.Flag(options.AuthMethodAuthCodeOption.Flag.Name)
+
+	flagProvided := (deviceCodeFlag != nil && deviceCodeFlag.Changed) ||
+		(clientCredentialsFlag != nil && clientCredentialsFlag.Changed) ||
+		(authCodeFlag != nil && authCodeFlag.Changed)
+
+	// If no flag was provided, check if authentication type is configured
+	if !flagProvided {
+		authType, err := profiles.GetOptionValue(options.PingOneAuthenticationTypeOption)
+		if err != nil || strings.TrimSpace(authType) == "" {
+			// No authentication type configured - run interactive setup
+			if err := auth_commands.RunInteractiveAuthConfig(os.Stdin); err != nil {
+				return err
+			}
+			// After interactive setup, re-read the auth type
+			authType, err = profiles.GetOptionValue(options.PingOneAuthenticationTypeOption)
+			if err != nil {
+				return &errs.PingCLIError{
+					Prefix: "failed to read authentication type after configuration",
+					Err:    err,
+				}
+			}
+		}
+
+		// Now proceed with login using the configured authentication type
+		return performLoginByConfiguredType(ctx, authType, profileName)
+	}
+
+	// Flag was provided - use the flag value to override any configuration
+	deviceCodeStr, err := profiles.GetOptionValue(options.AuthMethodDeviceCodeOption)
+	if err != nil {
+		return fmt.Errorf("failed to get device-code flag: %w", err)
+	}
+
+	clientCredentialsStr, err := profiles.GetOptionValue(options.AuthMethodClientCredentialsOption)
+	if err != nil {
+		return fmt.Errorf("failed to get client-credentials flag: %w", err)
+	}
+
+	// Determine which authentication method was requested
+	var selectedMethod string
+	switch {
+	case deviceCodeStr == "true":
+		selectedMethod = string(svcOAuth2.GrantTypeDeviceCode)
+	case clientCredentialsStr == "true":
+		selectedMethod = string(svcOAuth2.GrantTypeClientCredentials)
+	default:
+		selectedMethod = string(svcOAuth2.GrantTypeAuthCode)
+	}
+
+	// Check if configuration exists for the selected method, prompt for setup if missing
+	if err := ensureAuthConfigurationExists(selectedMethod); err != nil {
+		// Configuration is missing - run interactive setup
+		if configErr := auth_commands.RunInteractiveAuthConfig(os.Stdin); configErr != nil {
+			return configErr
+		}
+	}
+
+	// Perform authentication based on selected method flag
+	var token *oauth2.Token
+	var newAuth bool
+
+	switch selectedMethod {
```

Comment:
This block below the `if !flagProvided` block seems to duplicate the logic in `performLoginByConfiguredType`, we should avoid the repetition